### ISSUE TITLE
Investigate compiling all of classes.jar and midlet on first run.

### ISSUE THREAD

**Issue by brendandahl** [mozilla/j2me.js#1423]
One of the major issues I'm seeing with jitting code is the lag we introduce. We jit hot methods but then those in turn have to be jitted by the js engine when it deems that they are hot. 

At various points I've tried compiling all of classes.jar with a large midlet and it previously has lead to large startup regressions.  Recently I tried this again it and the startup performance improved on desktop and was slightly worse on device.

If we compile everything on first run I see there being a number of benefits: we could do more optimizations, have a lot less unwinding code, only load the compiler on first run, and we could focus more on the quality of the output js code and rely on the js engine to do it's job at optimizing things. Things we'd need to test:

- how long it takes to compile everything on a device
- possibly loading compiled code by class or method

**Comment by marco-c** [comment]
> how long it takes to compile everything on a device

This shouldn't be a big deal if we only do it on first run.

**Comment by mykmelez** [comment]
> This shouldn't be a big deal if we only do it on first run.

Right, we can explain this away via an "optimizing" step that we do only on first run (and after each midlet update).
